app/service/agent_svc.py

```diff
@@ -51,7 +51,7 @@ def __init__(self, untrusted_timer):
         await self.get_service('data_svc').update('core_agent', 'paw', paw, data)
         self.log.debug('Agent %s is now trusted: %s' % (paw, bool(int(trusted))))
 
-    async def handle_heartbeat(self, paw, platform, server, group, executors, architecture, location, pid, ppid, sleep):
+    async def handle_heartbeat(self, paw, hostname, username, platform, server, group, executors, architecture, location, pid, ppid, sleep):
```

Comment:
why store username and hostname? While useful information, unless we're using it elsewhere (I may just be missing it) I'd drop it out. the agent table is getting a bit "wide", which isn't a big concern but I'd veer toward including only the things we need to get ahead of a future problem (else we could create a _map table which contains agent data that we don't care to retrieve on every explode_agents call)

Reply:
looking at the rest of this PR, I think your rationale is because paw is now a random string. what if the paw change was simply adding a timestamp to the end of our existing paw? so it would be: host$user$time

thoughts?